lib/data/service/remote/auth/auth_service.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:asset_tracker/domain/entities/auth/user_login_entity.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+
+class AuthService {
```

Comment:
bu servisin ismini güncellemekte fayda var. Genel bir isim verilmiş fakat içeride firebase kullanılmış ayrıca herhangi bir abstracttan türetilmemiş. Şöyle yaparsak ilerisi için çok daha yararlı olur:

- AuthService adında bir abstract sınıf oluşturmak ve içerisine sadece metot imzasını vermek.
- Ardından bu sınıfın içindekileri FirebaseAuthService veya FirebaseAuthManager olarak yeni sınıfa aktarmak.
- IAuthRepository içerisinde de AuthService değişken olarak istemek.

Reply:
Reply > IAuthRepository içerisinde de AuthService değişken olarak istemek.

Domain katmaninda IAuthRepository deki imzalamarı data katmanındaki repositoryimde kullanıyorum.

Domaindeki İnterface class i datadaki service im de değişken olarak istesem ne gibi bir avantajı olur.

Sonuçta sadede o classta imzalama yapıyorum herhangi bir şekilde o service class ın instance ina ihtiyacı yok.

Reply:
Şöyle servis olarak istemekten ziyade auth servisi abstraction yapmak sana avantaj sağlayacak. Bir sonraki task olarak Google Auth veya telefonla giriş istesem nasıl geliştirme yapacaksın?

Reply:
Auth service i abstract interface class olarak tanımladım servis işlemini FirebaseAuthService adlı bir classi implement ederek yaptım.